hazelcast/src/main/java/com/hazelcast/client/config/ClientConfig.java

```diff
@@ -168,6 +172,45 @@ public ClientConfig(ClientConfig config) {
         metricsConfig = new ClientMetricsConfig(config.metricsConfig);
     }
 
+    /**
+     * Populates Hazelcast {@link ClientConfig} object from an external configuration file.
+     * <p>
+     * It tries to load Hazelcast Client configuration from a list of well-known locations.
+     * When no location contains Hazelcast Client configuration then it returns default.
+     * <p>
+     * Note that the same mechanism is used when calling
+     * {@link com.hazelcast.client.HazelcastClient#newHazelcastClient()}.
+     *
+     * @return ClientConfig created from a file when exists, otherwise default.
+     */
+    public static ClientConfig load() {
+        validateSuffixInSystemProperty(SYSPROP_CLIENT_CONFIG);
+
+        XmlClientConfigLocator xmlConfigLocator = new XmlClientConfigLocator();
+        YamlClientConfigLocator yamlConfigLocator = new YamlClientConfigLocator();
+
+        ClientConfig config;
+        if (xmlConfigLocator.locateFromSystemProperty()) {
+            // 1. Try loading XML config from the configuration provided in system property
+            config = new XmlClientConfigBuilder(xmlConfigLocator).build();
```

Comment:
100% flavor comment. Why not return immediately from the function?

Reply:
Good idea! Changed.